File: frontend/lib/models/booking.dart
import 'event.dart';

class Booking {
  final String id;
  final Event event;
  final String ticketType;
  final int quantity;
  final double totalPrice;
  final String paymentStatus;
  final String qrCodeToken;
  final bool checkInStatus;

  Booking({
    required this.id,
    required this.event,
    required this.ticketType,
    required this.quantity,
    required this.totalPrice,
    required this.paymentStatus,
    required this.qrCodeToken,
    required this.checkInStatus,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['_id'] ?? '',
      event: Event.fromJson(json['event'] ?? {}),
      ticketType: json['ticketType'] ?? '',
      quantity: json['quantity'] ?? 1,
      totalPrice: (json['totalPrice'] ?? 0).toDouble(),
      paymentStatus: json['paymentStatus'] ?? 'Pending',
      qrCodeToken: json['qrCodeToken'] ?? '',
      checkInStatus: json['checkInStatus'] ?? false,
    );
  }
}

File: backend/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  event: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Event',
    required: true
  },
  ticketType: {
    type: String,
    required: true
  },
  quantity: {
    type: Number,
    required: true
  },
  totalPrice: {
    type: Number,
    required: true
  },
  paymentStatus: {
    type: String,
    enum: ['Pending', 'Completed', 'Failed', 'Refunded', 'Paid'],
    default: 'Pending'
  },
  razorpayOrderId: {
    type: String
  },
  razorpayPaymentId: {
    type: String
  },
  qrCodeToken: {
    type: String,
    unique: true
  },
  checkInStatus: {
    type: Boolean,
    default: false
  }
}, { timestamps: true });

module.exports = mongoose.model('Booking', bookingSchema);
